src/Microsoft.DotNet.InternalAbstractions/DirectoryPath.cs

```diff
@@ -10,8 +10,17 @@ public struct DirectoryPath
     {
         public string Value { get; }
 
+        /// <summary>
+        /// Create DirectoryPath to repesent a absolute directory path. Note it may not exist.
+        /// </summary>
+        /// <param name="value">If the value is not rooted. Path.GetFullPath will be called during the consturctor.</param>
         public DirectoryPath(string value)
         {
+            if (!Path.IsPathRooted(value))
```

Comment:
I am not sure I should automatic convert to fullpath or throw on not rooted.

Both of them has no type check (design time feed back). And if your tests always use full path, it will throw on runtime when it is not.

It is more pure to throw on not rooted. But GetParentPath are calling full path already.